core/chains/evm/headtracker/head_tracker.go

```diff
@@ -39,25 +41,67 @@ var (
 // HeadsBufferSize - The buffer is used when heads sampling is disabled, to ensure the callback is run for every head
 const HeadsBufferSize = 10
 
-type headTracker struct {
+type headTracker[
+	HTH htrktypes.Head[BLOCK_HASH, ID],
+	S commontypes.Subscription,
+	ID txmgrtypes.ID,
+	BLOCK_HASH commontypes.Hashable,
+] struct {
 	log             logger.Logger
-	headBroadcaster httypes.HeadBroadcaster
-	headSaver       httypes.HeadSaver
+	headBroadcaster commontypes.HeadBroadcaster[HTH, BLOCK_HASH]
+	headSaver       commontypes.HeadSaver[HTH, BLOCK_HASH]
 	mailMon         *utils.MailboxMonitor
-	ethClient       evmclient.Client
-	chainID         big.Int
+	ethClient       htrktypes.Client[HTH, S, ID, BLOCK_HASH]
+	chainID         ID
 	config          htrktypes.Config
 
-	backfillMB   *utils.Mailbox[*evmtypes.Head]
-	broadcastMB  *utils.Mailbox[*evmtypes.Head]
-	headListener commontypes.HeadListener[*evmtypes.Head, common.Hash]
+	backfillMB   *utils.Mailbox[HTH]
+	broadcastMB  *utils.Mailbox[HTH]
+	headListener commontypes.HeadListener[HTH, BLOCK_HASH]
 	chStop       utils.StopChan
 	wgDone       sync.WaitGroup
 	utils.StartStopOnce
+	getNilHead func() HTH
 }
 
+type evmHeadTracker = headTracker[*evmtypes.Head, ethereum.Subscription, *big.Int, common.Hash]
+
+var _ commontypes.HeadTracker[*evmtypes.Head, common.Hash] = (*evmHeadTracker)(nil)
+
 // NewHeadTracker instantiates a new HeadTracker using HeadSaver to persist new block numbers.
-func NewHeadTracker(
+func NewHeadTracker[
+	HTH htrktypes.Head[BLOCK_HASH, ID],
+	S commontypes.Subscription,
+	ID txmgrtypes.ID,
+	BLOCK_HASH commontypes.Hashable,
+](
+	lggr logger.Logger,
+	client htrktypes.Client[HTH, S, ID, BLOCK_HASH],
+	config htrktypes.Config,
+	headBroadcaster commontypes.HeadBroadcaster[HTH, BLOCK_HASH],
+	headSaver commontypes.HeadSaver[HTH, BLOCK_HASH],
+	mailMon *utils.MailboxMonitor,
+	getNilHead func() HTH,
+) commontypes.HeadTracker[HTH, BLOCK_HASH] {
+	chStop := make(chan struct{})
+	lggr = lggr.Named("HeadTracker")
+	return &headTracker[HTH, S, ID, BLOCK_HASH]{
+		headBroadcaster: headBroadcaster,
+		ethClient:       client,
+		chainID:         client.ConfiguredChainID(),
+		config:          config,
+		log:             lggr,
+		backfillMB:      utils.NewSingleMailbox[HTH](),
+		broadcastMB:     utils.NewMailbox[HTH](HeadsBufferSize),
+		chStop:          chStop,
+		headListener:    NewHeadListener[HTH, S, ID, BLOCK_HASH](lggr, client, config, chStop),
+		headSaver:       headSaver,
+		mailMon:         mailMon,
+		getNilHead:      getNilHead,
+	}
+}
+
+func NewEvmHeadTracker(
```

Comment:
Why change this name? Seems redundant since we already know this package is EVM specific, no?

Related: We should be using standard naming `EVM` everywhere, instead of `Evm`.

Reply:
Ya, till the generic code is currently in this same file, it makes sense to have NewEVMHeadTracker() name.
But I agree with Jordan, that once the generic code is moved to the common/... path, then all code inside the .../evm/... path need not repeat EVM prefix.
I see @aalu1418 already opened a ticket to do it for TXM. We should follow the same pattern here too.
I'm ok if you want to do the refactor and renames separately.

Reply:
Will be opening a ticket to do the cleanup as well for Headtracker.